packages/block-library/src/navigation-menu/block-colors-selector.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { IconButton, Dropdown, Toolbar } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { DOWN } from '@wordpress/keycodes';
+import { ColorPaletteControl, ContrastChecker } from '@wordpress/block-editor';
+
+/**
+ * Color Selector Icon component.
+ * @return {*} React Icon component.
+ * @constructor
+ */
+const ColorSelectorIcon = ( { style } ) =>
+	<div className="block-editor-block-colors-selector__icon-container">
+		<div
+			className="colors-selector__state-selection wp-block-navigation-menu-item"
+			style={ style }
+		>
+			{ __( 'Aa' ) }
+		</div>
+	</div>;
+
+/**
+ * Renders the Colors Selector Toolbar with the icon button.
+ *
+ * @param {function} onToggle Callback open/close Dropdown.
+ * @param {bool} isOpen True is the color settings dropdown is open. Otherwise, False.
+ * @return {*} React toggle button component.
+ */
+const renderToggle = ( style ) => ( { onToggle, isOpen } ) => {
+	const openOnArrowDown = ( event ) => {
+		if ( ! isOpen && event.keyCode === DOWN ) {
+			event.preventDefault();
+			event.stopPropagation();
+			onToggle();
+		}
+	};
+
+	return (
+		<Toolbar>
+			<IconButton
+				className="components-icon-button components-toolbar__control block-editor-block-colors-selector__toggle"
+				label={ __( 'Open Colors Selector' ) }
+				onClick={ onToggle }
+				onKeyDown={ openOnArrowDown }
+				icon={ <ColorSelectorIcon style={ style }/> }
+			/>
+		</Toolbar>
+	);
+};
+
+const renderContent = ( { backgroundColor, textColor, setBackgroundColor, setTextColor } ) => ( ( { isOpen, onToggle, onClose } ) => {
+	return (
+		<>
+			<div className="color-palette-controller-container">
+				<ColorPaletteControl
+					value={ backgroundColor.color }
```

Comment:
Just checking...does `backgroundColor` **_always_** have a valid `color` property?

Reply:
Yes, I guess so 😅 
I've tested renaming the prop name to som random value and in the worse of cases it just won't show the ✔️ in the circle once the color is selected. Something similar happens with the colors checker. Althought they won't work as expected won't break the app neither.
Just FYI, these `backgroundColor` and `textColors` are properties created with the `withColors` HOC and populated from the `<NavigationMenu />` component.

```es6
export default compose( [
	withColors( { backgroundColor: 'background-color', textColor: 'color' } ),
	// ...
] )( NavigationMenu );
```